lib/jekyll/filters/url_filters.rb

```diff
@@ -20,13 +20,17 @@ def absolute_url(input)
         ).normalize.to_s
       end
 
-      # Produces a URL relative to the domain root based on site.baseurl.
+      # Produces a URL relative to the domain root based on site.baseurl
+      # unless it is already an absolute url with an authority (host).
       #
       # input - the URL to make relative to the domain root
       #
       # Returns a URL relative to the domain root as a String.
       def relative_url(input)
         return if input.nil?
+        uri = Addressable::URI.parse(input.to_s)
+        return uri.normalize.to_s if uri.absolute?
```

Comment:
What does `normalize` actually do? Do we need that?

Reply:
By which I mean: can we instead `return input` instead?

Reply:
Oh, my bad. I thought `absolute_url` was using it for already absolute URIs. But in fact it normalizes only those generated by that method. So, `relative_url` should behave the same way.